app/src/main/java/rightmesh/left/io/gpstracker/utils/PermissionUtil.java

```diff
@@ -0,0 +1,151 @@
+
+    public PermissionUtil(Activity activity) {
+        this.activity = activity;
+    }
+
+    /**
+     * Add permission that need to be requested.
+     *
+     * @param permissions list of permission to ask for
+     * @return @{@link PermissionUtil}
+     */
+    public PermissionUtil addPermissions(String... permissions) {
+        this.permissions = permissions;
+        return this;
+    }
+
+    /**
+     * Add callback to handle logic when permissions are granted or denied.
+     *
+     * @param callback provided
+     * @return {@link PermissionUtil}
+     */
+    public PermissionUtil callback(PermissionCallback callback) {
+        this.callback = callback;
+        return this;
+    }
+
+    /**
+     * Add a lifecyle so {@link PermissionUtil} will know when Activity/Fragment is destroyed.
+     *
+     * @param lifecycle to observe lifecycle
+     * @return {@link PermissionUtil}
+     */
+    public PermissionUtil addLifeCycleOwner(Lifecycle lifecycle) {
+        this.lifecycle = lifecycle;
+        if (this.lifecycle != null) {
+            this.lifecycle.addObserver(this);
+        }
+        return this;
+    }
+
+    /**
+     * Check if permissions are all granted.
+     *
+     * @return true: all permissions are granted.<br>
+     * false: one of permissions is denied.
+     */
+    private boolean isPermissionAllGranted() {
+        for (String permission : permissions) {
+            if (ActivityCompat.checkSelfPermission(activity, permission) != PERMISSION_GRANTED) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Requesting permission.
+     */
+    public void check() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !isPermissionAllGranted()) {
+            ActivityCompat.requestPermissions(activity, permissions, PERMISSION_REQUEST_CODE);
+        } else {
+            if (callback != null) {
+                callback.onAllGranted();
+            }
+        }
+    }
+
+    /**
+     * Always put this method in {@link Activity#onRequestPermissionsResult(int, String[], int[])}
+     * or {@link android.support.v4.app.Fragment#onRequestPermissionsResult(int, String[], int[])}
+     * to handle result of permission requesting.<br>
+     * Pass the same provided params in onRequestPermissionsResult() to this method
+     *
+     * @param requestCode  The request code passed in.
+     * @param permissions  The requested permissions. Never null.
+     * @param grantResults The grant results for the corresponding permissions
+     */
+    public void handleResult(int requestCode,
+                             @NonNull String[] permissions,
+                             @NonNull int[] grantResults) {
+        if (requestCode == PERMISSION_REQUEST_CODE) {
+            ArrayList<String> grantedPermissions = new ArrayList<>();
+            ArrayList<String> deniedPermission = new ArrayList<>();
+
+            for (int i = 0; i < permissions.length; i++) {
```

Comment:
I believe you can do `for (String permission: permissions)` even on primitive arrays, if you don't want to have to deal with indices.